code/modules/projectiles/projectile/special/spidernet.dm

```diff
@@ -0,0 +1,41 @@
+/obj/item/projectile/bullet/spidernet
+	name = "sticky webbing"
+	icon_state = "spidernet"
+	damage = 0
+
+/obj/item/projectile/bullet/spidernet/on_hit(atom/target, blocked = FALSE)
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		C.Knockdown(4 SECONDS)
+	else if(isliving(target)) //we do NOT want to apply this effect to carbons
+		var/mob/living/L = target
+		L.Immobilize(4 SECONDS)
+	return ..()
+
+/obj/item/projectile/bullet/spidernet/on_hit(atom/target, blocked = FALSE)
+	. = ..()
+	if(isliving(target))
+		var/turf/T = get_turf(target)
+		web_tile(T)
+	else
+		web_tile()
+/obj/item/projectile/bullet/spidernet/on_range()
+	web_tile()
```

Comment:
what is going on here with on_hit and then why is web_tile() called here as well

Reply:
`on_range()` is called when the projectile ends. `on_hit()` is called when the projectile collides with something, whether that's a wall, spessman or otherwise.

`web_tile()` is called because regardless of why the projectile's life has ended, I want it to web the tile it is currently on.